Docs: Transfer in-memory-web-api into main repo

SijmenHuizenga opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
In #37182 the in-memory-web-api module was moved into this repo. The documentation has not yet been updated accordingly. There are two problems:

* At the moment the aio docs still have some links to the archived repo: `aio/content/guide/http.md` and `aio/content/tutorial/toh-pt6.md`
* The readme is still only available in the old repo.

Issue Number: Fixes #40190

## What is the new behavior?

Make links in the docs point to the new location of the in-memory-web-api.

Copy the reamde into this repo with the following changes:
 * Removed Travis badges
 * Updated github links o piont to angular repo
 * Removed 'running tests' as it no longer relevant



## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

gkalpak commented on packages/misc/angular-in-memory-web-api/README.md:
```diff
@@ -0,0 +1,386 @@
+
+Format: `/app/heroes/?propertyName=regexPattern`
+
+The following example matches all names start with the letter 'j'  or 'J' in the heroes collection.
```
```suggestion
The following example matches all names that start with the letter 'j' or 'J' in the heroes collection.
```

gkalpak reviewed (changes requested):
Thx, @SijmenHuizenga! I've left a few minor comments. Otherwise lgtm 🚀

BTW, there are some typos in the commit messages:
- `links o piont to` --> `links to point to`
- `as it no longer` --> `as it is no longer`
- `Fixes: #40190` --> `Fixes #40190` (in both commits)

gkalpak commented:
I forgot to mention, I think it is a good idea to include a note in the README that says this project used to live at https://github.com/angular/in-memory-web-api.

SijmenHuizenga commented:
Thanks a lot for the corrections and suggestions @gkalpak! I've applied them all. Let me know if there is anything else you would like to see changed.

IgorMinar reviewed (approved):
LGTM. Thank you!

IgorMinar commented:
@SijmenHuizenga  can you please rebase this PR and force push to your branch? thanks

SijmenHuizenga committed:
docs: import in-memory-web-api readme

In #37182 the in-memory-web-api module was moved into this repo.
Copy the reamde into this repo with the following changes:
 * Removed Travis badges
 * Updated github links to point to Angular repo
 * Removed 'running tests' as it is no longer relevant

Fixes #40190

SijmenHuizenga committed:
docs: change links to in-memory-web-api in github/angular repo

In #37182 the in-memory-web-api module was moved into this repo.
Make links in the docs point to the new location of the in-memory-web-api.

Fixes #40190

SijmenHuizenga committed:
fixup! docs: import in-memory-web-api readme

gkalpak committed:
fixup! docs: change links to in-memory-web-api in github/angular repo

gkalpak commented:
@googlebot I consent.
